lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -597,10 +638,21 @@ void LowerLayersPass::runOnOperation() {
   llvm::sys::SmartMutex<true> mutex;
   circuitMutex = &mutex;
 
+  // Extract information from circuit-level annotations.
+  for (auto anno : AnnotationSet(circuitOp)) {
+    if (!anno.isClass(testBenchDirAnnoClass))
+      continue;
+    auto dir = anno.getMember<StringAttr>("dirname");
+    assert(dir && "invalid test bench annotation");
+    testBenchDir = dir.getValue();
+  }
+
   // Determine names for all modules that will be created.  Do this serially to
-  // avoid non-determinism from creating these in the parallel region.
+  // avoid non-determinism from creating these in the parallel region.  While
+  // walking modules, set the DUT if it is so marked.
   CircuitNamespace ns(circuitOp);
   circuitOp->walk([&](FModuleOp moduleOp) {
+    (void)extractDUT(moduleOp, dut);
```

Comment:
What's going on here -- why ignoring the failure (which looks like will print a diagnostic)?

Reply:
I thought there was a reason that this could be ignored here. However, I don't think that's right. I'll add the check.

Reply:
I think my original reasoning here was that `extractDUT` is used and checked elsewhere in the pipeline, so erroring here, too, was weird. That said, passes should work standalone and not require a pass pipeline ordering. Reiterating that the check is now in there.

Reply:
Changed to signal pass failure in: https://github.com/llvm/circt/pull/6741/commits/6264b954a75864af329dae84b995c12910664452 A test is included in: https://github.com/llvm/circt/pull/6741/commits/fb80c1b3b6023aa949188f1696712aee77a65e5a.